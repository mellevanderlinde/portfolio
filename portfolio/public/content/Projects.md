## API Gateway Integration with Amazon Bedrock
Small AWS CDK project that deploys an API endpoint to interact with a large language model from Amazon's generative AI service.
- ![API Gateway and Bedrock](../assets/bedrock-apigateway.png)
- Tags: Category 1
- Badges:
  - AWS [orange]
  - TypeScript [green]
- Buttons:
  - GitHub Repo [https://github.com/mellevanderlinde/bedrock-apigateway]

## Project B
Description
- ![600x200](https://via.placeholder.com/600x200)
- Tags: Category 2
- Badges:
  - Badge [blue]
- Buttons:
  - Link [https://example.com]