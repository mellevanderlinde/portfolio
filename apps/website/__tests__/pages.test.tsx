import Blog from '@app/blog/page'
import NotFound from '@app/not-found'
import Home from '@app/page'
import Projects from '@app/projects/page'
import Work from '@app/work/page'
import { render } from '@testing-library/react'
import { expect, it } from 'vitest'

it('matches home', () => {
  const { container } = render(<Home />)
  expect(container).toMatchSnapshot()
})

it('matches blog', () => {
  const { container } = render(<Blog />)
  expect(container).toMatchSnapshot()
})

it('matches projects', () => {
  const { container } = render(<Projects />)
  expect(container).toMatchSnapshot()
})

it('matches work', () => {
  const { container } = render(<Work />)
  expect(container).toMatchSnapshot()
})

it('matches not found', () => {
  const { container } = render(<NotFound />)
  expect(container).toMatchSnapshot()
})
