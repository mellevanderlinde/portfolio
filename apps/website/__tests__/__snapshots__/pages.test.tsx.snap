// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`matches blog 1`] = `
<div>
  <section>
    <h1
      class="mb-8 text-2xl font-medium tracking-tight"
    >
      Blog
    </h1>
    <div>
      <a
        class="flex flex-col space-y-1 mb-4 transition-opacity duration-200 hover:opacity-80"
        href="/blog/openid-connect-aws-github"
      >
        <div
          class="w-full flex flex-col sm:flex-row justify-between items-start sm:items-center space-y-1 sm:space-y-0 sm:space-x-2"
        >
          <p
            class="text-black dark:text-white tracking-tight"
          >
            Connect GitHub Actions with AWS using OpenID Connect
          </p>
          <p
            class="text-neutral-600 dark:text-neutral-400 tabular-nums text-sm"
          >
            Sep 26, 2024
          </p>
        </div>
      </a>
    </div>
  </section>
</div>
`;

exports[`matches home 1`] = `
<div>
  <section>
    <h1
      class="mb-8 text-2xl font-medium tracking-tight"
    >
      My Portfolio
    </h1>
    <div
      class="prose prose-neutral dark:prose-invert"
    >
      <p>
        I have an interest in cloud engineering, machine learning, technology and sustainability. After completing my Master's in Behavioural Data Science, I joined PostNL as MLOps Engineer, where I now work as Cloud Engineer.
      </p>
    </div>
  </section>
</div>
`;

exports[`matches not found 1`] = `
<div>
  <section>
    <h1
      class="mb-8 text-2xl font-medium tracking-tight"
    >
      404 - Page not found
    </h1>
    <p
      class="mb-4"
    >
      The page you are looking for does not exist.
    </p>
  </section>
</div>
`;

exports[`matches projects 1`] = `
<div>
  <section>
    <h1
      class="mb-8 text-2xl font-medium tracking-tight"
    >
      Projects
    </h1>
    <div
      class="space-y-6"
    >
      <a
        class="block group hover:opacity-80 transition-opacity duration-200"
        href="https://github.com/mellevanderlinde/energy-meter"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="flex flex-col"
        >
          <div
            class="w-full flex justify-between items-baseline"
          >
            <span
              class="text-black dark:text-white font-medium tracking-tight"
            >
              Energy Data on AWS with Raspberry Pi
            </span>
            <span
              class="text-neutral-600 dark:text-neutral-400 tabular-nums text-sm"
            >
              2025
            </span>
          </div>
          <p
            class="prose prose-neutral dark:prose-invert pt-3"
          >
            Creates a service on a Raspberry Pi to store home energy usage data on DynamoDB by reading the P1 port.
          </p>
        </div>
      </a>
      <a
        class="block group hover:opacity-80 transition-opacity duration-200"
        href="https://github.com/mellevanderlinde/portfolio"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="flex flex-col"
        >
          <div
            class="w-full flex justify-between items-baseline"
          >
            <span
              class="text-black dark:text-white font-medium tracking-tight"
            >
              Portfolio with CloudFront and Next.js
            </span>
            <span
              class="text-neutral-600 dark:text-neutral-400 tabular-nums text-sm"
            >
              2024
            </span>
          </div>
          <p
            class="prose prose-neutral dark:prose-invert pt-3"
          >
            AWS CDK project that serves this Next.js portfolio with Amazon CloudFront. This portfolio design is a template by Vercel.
          </p>
        </div>
      </a>
      <a
        class="block group hover:opacity-80 transition-opacity duration-200"
        href="https://github.com/mellevanderlinde/oidc-aws-github"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="flex flex-col"
        >
          <div
            class="w-full flex justify-between items-baseline"
          >
            <span
              class="text-black dark:text-white font-medium tracking-tight"
            >
              OpenID Connect for AWS and GitHub Actions
            </span>
            <span
              class="text-neutral-600 dark:text-neutral-400 tabular-nums text-sm"
            >
              2024
            </span>
          </div>
          <p
            class="prose prose-neutral dark:prose-invert pt-3"
          >
            This repository demonstrates how to allow a specific GitHub repository to invoke actions on AWS from GitHub Actions using OpenID Connect.
          </p>
        </div>
      </a>
      <a
        class="block group hover:opacity-80 transition-opacity duration-200"
        href="https://github.com/aws/aws-cdk/commits/main/?author=mellevanderlinde"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="flex flex-col"
        >
          <div
            class="w-full flex justify-between items-baseline"
          >
            <span
              class="text-black dark:text-white font-medium tracking-tight"
            >
              Contributions to AWS CDK
            </span>
            <span
              class="text-neutral-600 dark:text-neutral-400 tabular-nums text-sm"
            >
              2024
            </span>
          </div>
          <p
            class="prose prose-neutral dark:prose-invert pt-3"
          >
            This link shows some contributions (tiny, mostly bug fixes) to the AWS CDK repository.
          </p>
        </div>
      </a>
      <a
        class="block group hover:opacity-80 transition-opacity duration-200"
        href="https://github.com/mellevanderlinde/mwaa-airflow"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="flex flex-col"
        >
          <div
            class="w-full flex justify-between items-baseline"
          >
            <span
              class="text-black dark:text-white font-medium tracking-tight"
            >
              Apache Airflow on AWS
            </span>
            <span
              class="text-neutral-600 dark:text-neutral-400 tabular-nums text-sm"
            >
              2024
            </span>
          </div>
          <p
            class="prose prose-neutral dark:prose-invert pt-3"
          >
            AWS CDK project that deploys an MWAA environment for Apache Airflow. The repository contains a minimal DAG that invokes a Lambda function.
          </p>
        </div>
      </a>
      <a
        class="block group hover:opacity-80 transition-opacity duration-200"
        href="https://github.com/mellevanderlinde/bedrock-apigateway"
        rel="noopener noreferrer"
        target="_blank"
      >
        <div
          class="flex flex-col"
        >
          <div
            class="w-full flex justify-between items-baseline"
          >
            <span
              class="text-black dark:text-white font-medium tracking-tight"
            >
              Bedrock API
            </span>
            <span
              class="text-neutral-600 dark:text-neutral-400 tabular-nums text-sm"
            >
              2023
            </span>
          </div>
          <p
            class="prose prose-neutral dark:prose-invert pt-3"
          >
            AWS CDK project that deploys an API endpoint using API Gateway. The endpoint interacts with a large language model from Amazon's generative AI service.
          </p>
        </div>
      </a>
    </div>
  </section>
</div>
`;

exports[`matches work 1`] = `
<div>
  <section>
    <h1
      class="mb-8 text-2xl font-medium tracking-tight"
    >
      Work
    </h1>
    <div
      class="space-y-6"
    >
      <div
        class="flex flex-col"
      >
        <div
          class="w-full flex justify-between items-baseline"
        >
          <span
            class="text-black dark:text-white font-medium tracking-tight"
          >
            Cloud Engineer
             
            @
             
            PostNL
          </span>
          <span
            class="text-neutral-600 dark:text-neutral-400 tabular-nums text-sm"
          >
            2023
             
            -
             
            Present
          </span>
        </div>
        <p
          class="prose prose-neutral dark:prose-invert pt-3"
        >
          In 2023 I joined the Cloud Center of Excellence (CCoE) team. My (platform) 
      team works on the serverless cloud platform on AWS, used by all engineers. 
      The stack includes a wide variety of services from AWS (e.g., IAM, 
      CloudFormation, API Gateway, Organizations, Lambda, DynamoDB).
        </p>
        <p
          class="prose prose-neutral dark:prose-invert pt-3"
        >
          As an example project, I guided engineering teams to migrate their GitHub 
      Actions to use OpenID Connect (OIDC) for AWS and JFrog integrations. This 
      project uses an internal tool I developed to make OIDC integration easier for engineers.
      The new solution is more secure, easier to maintain and more cost-effective.
        </p>
      </div>
      <div
        class="flex flex-col"
      >
        <div
          class="w-full flex justify-between items-baseline"
        >
          <span
            class="text-black dark:text-white font-medium tracking-tight"
          >
            MLOps Engineer
             
            @
             
            PostNL
          </span>
          <span
            class="text-neutral-600 dark:text-neutral-400 tabular-nums text-sm"
          >
            2021
             
            -
             
            2023
          </span>
        </div>
        <p
          class="prose prose-neutral dark:prose-invert pt-3"
        >
          Responsible for the deployment and monitoring of data science
        models on AWS. The stack included containerization (Docker), 
        infrastructure as code (AWS CDK), CI/CD (GitHub Actions and AWS 
        CodePipeline), orchestration (Airflow), parallelization (PySpark).
        </p>
      </div>
    </div>
  </section>
</div>
`;
